webui/react/src/components/ResourcePoolCard.tsx

```diff
@@ -115,7 +114,9 @@ const ResourcePoolCard: React.FC<Props> = ({
   }, [pool]);
 
   const processedPool = useMemo(() => {
-    const newPool = clone(pool);
+    // had to add this in order to avoid adding ts-ignore at lines 121 and 122 due to incompatibility of types
+    // since this is sort of a translation of the API response object into UI, this should be fine.
+    const newPool = JSON.parse(JSON.stringify(structuredClone(pool)));
```

Comment:
given that `pool` is an api response and we're not mutating the object, i'm wondering if using `structuredClone` is the right approach here.
we could instead do something like:

```tsx
const { slotsPerAgent, schedulerType } = pool;
return {
  ...pool,
  slotsPerAgent: slotsPerAgent === -1 ? 'Unknown' : slotsPerAgent,
  schedulerType: V1SchedulerTypeToLabel[schedulerType]
}
```